tuf/signed/sign_test.go

```diff
@@ -264,9 +267,87 @@ func TestSignWithX509(t *testing.T) {
 	// RSAx509 key)
 	mockCryptoService := &StrictMockCryptoService{MockCryptoService{privKey}}
 	testData := data.Signed{}
-	err = Sign(mockCryptoService, &testData, tufRSAx509Key)
+	baseRole := data.NewBaseRole(
+		data.CanonicalRootRole,
+		1,
+		tufRSAx509Key,
+	)
+	err = Sign(mockCryptoService, &testData, baseRole)
 	assert.NoError(t, err)
 
 	assert.Len(t, testData.Signatures, 1)
 	assert.Equal(t, tufRSAx509Key.ID(), testData.Signatures[0].KeyID)
 }
+
+func TestSignWithOldKey(t *testing.T) {
+	cs := NewEd25519()
+	testData := data.Signed{}
+
+	key1, err := cs.Create("root", data.ED25519Key)
```

Comment:
nit: can we change this and the other `"root"` strings to `data.CanonicalRootRole`?